homework/index.js

```diff
@@ -1,46 +1,99 @@
+    if (err) {
+      createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+      return;
+    }
+    const repNames = [];
+    createAndAppend('select', root, { id: 'main-select' });
+    createAndAppend('div', root, { class: 'portion-parent', id: 'parent' });
+    const parent = document.getElementById('parent');
+    createAndAppend('div', parent, { class: 'left-portion', id: 'lft' });
+    createAndAppend('div', parent, { class: 'right-portion', id: 'rt' });
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
+    const leftPortion = document.getElementById('lft');
+    const rightPortion = document.getElementById('rt');
+    for (let i = 0; i < repositories.length; i++) {
+      const rep = repositories[i];
+      repNames.push(rep.name);
+    }
+    const selectItem = document.getElementById('main-select');
+
+    repNames.sort();
+    for (let i = 0; i < repNames.length; i++) {
+      createAndAppend('option', selectItem, {
+        value: i,
+        text: repNames[i],
+      });
+    }
+    selectItem.selectedIndex = 0;
+    selectItem.addEventListener('change', () => {
+      const reponame = selectItem.options[selectItem.selectedIndex].text;
+      for (let i = 0; i < repositories.length; i++) {
+        if (repositories[i].name === reponame) {
+          const repo = repositories[i];
+          leftPortion.innerHTML = '';
+          createAndAppend('p', leftPortion, { text: repo.name });
+          createAndAppend('p', leftPortion, { text: repo.description });
+          createAndAppend('p', leftPortion, { text: repo.forks });
+          createAndAppend('p', leftPortion, { text: repo.updated_at });
+          const contUrl = repo.contributors_url;
+          fetchJSON(contUrl, (cerr, contList) => {
+            rightPortion.innerHTML = '';
+            for (let j = 0; j < contList.length; j++) {
+              createAndAppend('div', rightPortion, {
+                class: 'rightParent',
+                id: `contList${[j]}`,
+              });
+              const rightParent = document.getElementById(`contList${j}`);
+              createAndAppend('img', rightParent, {
+                src: contList[j].avatar_url,
+              });
+              createAndAppend('p', rightParent, { text: contList[j].login });
+              createAndAppend('p', rightParent, {
+                text: contList[j].contributions,
+              });
+            }
+          });
+        }
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
     });
-  }
 
-  const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
-  window.onload = () => main(HYF_REPOS_URL);
+    // createAndAppend('pre', root, {text: JSON.stringify(repositories, null, 2)});
+  });
```

Comment:
The assignment specifies that, at application startup, you should render the details for the first repo in the list (the one that is displayed in the select box). This does not happen in your case. Note that the `change` event is only fired if the user actively changes the selection. Therefore, at application startup, you must manually run the same code that the event listener runs. You now have a big arrow function as the event handler which makes it difficult to run the same code from another place. There you should try and create a separate functoin that contains this code that you can call both from the even t listener and from another place to render the first repo at startup.